.github/workflows/ci.yml

```diff
@@ -9,6 +9,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Do we need it when running tests? Similarly for the token, now I come to think of it.

Reply:
We do need the token, as we run a full build as part of the test suite (to catch things like unparseable markdown, which should error the build).

We don't strictly need the `fetch-depth: 0` for the tests but I'm wary of the test and build environments diverging (and causing exactly the kind of bug this commit is trying to fix!).

Reply:
Ah, true. I think it's OK to digress here: we do benefit from speeding up the build, and it's highly unlikely that we'll be writing a test that depends on previous commits (since a future commit may always supersede them!).

Reply:
Agreed the risk is low 👍 I may raise a follow-up PR at some point, concentrating on bringing the build time down more generally.